[AAP] kubernetes docs for nginx (#31626)
DataDog/documentation: content/en/security/application_security/setup/nginx/kubernetes.md

The following Helm values demonstrates how to inject `nginx-datadog` module into an ingress-nginx controller:

```yaml
controller:
  config: 
    main-snippet: "load_module /modules_mount/ngx_http_datadog_module.so;"
  opentelemetry:
    enabled: false
  extraModules:
    - name: nginx-datadog
      image:
        registry: docker.io
        image: datadog/ingress-nginx-injection
        # The tag should match the version of the ingress-nginx controller
        # For example, this will inject the Datadog module for ingress v1.10.0
        # Check <https://hub.docker.com/repository/docker/datadog/ingress-nginx-injection/tags> 
        # for the list of all versions supported.
        tag: "v1.10.0"
        distroless: false
```

Check [our details examples](./examples/ingress-nginx) to help you set up ingress-nginx with `nginx-datadog`.

## How does it work?
Init containers are special containers that run before the main container in a Kubernetes pod. In this case,
the Datadog init container is responsible for copying the `nginx-datadog` module into a shared volume that will be
accessible by the main ingress-nginx container.

> When the main ingrees-nginx controller starts, the NGINX configuration must be updated with the `load_module` directive,

Reviewer: ```suggestion
When the main ingrees-nginx controller starts, the nginx configuration must be updated with the `load_module` directive,
```

allowing it to load the Datadog module seamlessly.

> > [!WARNING]  
> > We provide a specific init container **for each ingress-nginx controller version** start with `v1.10.0`. 
> > This is crucial because **each** init container must match the underlying NGINX version. Be sure to choose
> > the version of the Datadog init container matching your ingress-nginx version to ensure compatibility.

Reviewer: ```suggestion
<div class="alert alert-warning">
We provide a specific init container **for each ingress-nginx controller version**, starting with <code>v1.10.0</code>. This is crucial because **each** init container must match the underlying nginx version. To ensure compatibility, ensure the version of the Datadog init container matches your ingress-nginx version.
</div>
```

## Interaction with OpenTelemetry

> By default, ingress-nginx includes an OpenTelemetry (oTel) module, which can be enabled via the `enable-opentelemetry: true` setting
> in the [ingress-nginx ConfigMap](https://kubernetes.github.io/ingress-nginx/user-guide/nginx-configuration/configmap/#enable-opentelemetry).
> However, if you are using `nginx-datadog` for tracing, we recommend **disabling** OpenTelemetry to prevent duplicate tracing data from both
> the oTel and Datadog modules.

Reviewer: ```suggestion
By default, ingress-nginx includes an OpenTelemetry (oTel) module that can be enabled using the `enable-opentelemetry: true` setting
in the [ingress-nginx ConfigMap](https://kubernetes.github.io/ingress-nginx/user-guide/nginx-configuration/configmap/#enable-opentelemetry).
However, if you are using `nginx-datadog` for tracing, we recommend **disabling** OpenTelemetry to prevent duplicate tracing data from both
the oTel and Datadog modules.
```
